pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -148,6 +148,10 @@ public long getLatestIngestionTimestamp() {
     _segmentPartitionConfig = config.getSegmentPartitionConfig();
     _nullHandlingEnabled = config.isNullHandlingEnabled();
 
+    // Ensure that the schema has the virtual columns added
```

Comment:
Now virtual columns are only added to immutable segments, although in mutable segments there are logics in `getDataSource` for virtual columns. So adding virtual columns to mutable segments too.

Reply:
Virtual column is already added to the schema in RealtimeTableDataManager